code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -7,17 +7,89 @@
 	ignored_by = list(/mob/living/simple_animal/hostile/faithless)
 	meat = /obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/shadow
 	species_traits = list(NOBREATH,NOBLOOD,RADIMMUNE,VIRUSIMMUNE)
+
 	dangerous_existence = 1
 	mutanteyes = /obj/item/organ/eyes/night_vision
 
 
 /datum/species/shadow/spec_life(mob/living/carbon/human/H)
-	var/light_amount = 0
-	if(isturf(H.loc))
-		var/turf/T = H.loc
-		light_amount = T.get_lumcount()
+	var/turf/T = H.loc
+	if(istype(T))
+		var/light_amount = T.get_lumcount()
 
-		if(light_amount > 0.2) //if there's enough light, start dying
+		if(light_amount > SHADOW_SPECIES_LIGHT_THRESHOLD) //if there's enough light, start dying
 			H.take_overall_damage(1,1)
-		else if (light_amount < 0.2) //heal in the dark
+		else if (light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD) //heal in the dark
 			H.heal_overall_damage(1,1)
+
+
+/datum/species/shadow/nightmare
+	name = "Nightmare"
+	id = "nightmare"
+	limbs_id = "shadow"
+	burnmod = 1.5
+	blacklisted = TRUE
+	no_equip = list(slot_wear_mask, slot_wear_suit, slot_gloves, slot_shoes, slot_w_uniform, slot_s_store)
+	species_traits = list(NOBREATH,RESISTCOLD,RESISTPRESSURE,NOGUNS,NOBLOOD,RADIMMUNE,VIRUSIMMUNE,PIERCEIMMUNE,NODISMEMBER,NO_UNDERWEAR,NOHUNGER)
+	mutanteyes = /obj/item/organ/eyes/night_vision/nightmare
+
+/datum/species/shadow/nightmare/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	var/obj/effect/proc_holder/spell/targeted/shadowwalk/SW = new

```

Comment:
These need to be removed on species change. Also there was specific variable for species inhands. Zombies used that, you should probably use/repurpose that if possible.